Sources/Extensions/SwiftStdlib/SequenceExtensions.swift

```diff
@@ -135,6 +135,27 @@ public extension Sequence {
         })
     }
 
+    /// SwifterSwift: Get the only element based on optional condition.
+    ///
+    ///     [].single() -> nil
+    ///     [4].single() -> 4
+    ///     [2, 4].single() -> nil
+    ///     [1, 4, 7].single(where: {$0 % 2 == 0}) -> 4
+    ///     [2, 2, 4, 7].single(where: {$0 % 2 == 0}) -> nil
+    ///
+    /// - Parameter condition: condition to evaluate each element against (default is `{_ in true}`).
+    /// - Returns: The only element in the array matching the specified condition. If there are more matching elements, nil is returned. (optional)
+    public func single(where condition: ((Element) throws -> Bool) = { _ in true }) -> Element? {
+        var singleElement: Element?
+        for element in self where (try? condition(element)) ?? false {
+            guard singleElement == nil else {
```

Comment:
Yes, if you need to conditionally change scope then you should be using `guard`. 
It's not because it _shows intent_, it's because the compiler _will force you_ to exit the current scope. This is extra safety from the compiler.